source/common/http/http2/metadata_decoder.cc

```diff
@@ -0,0 +1,111 @@
+#include "common/http/http2/metadata_decoder.h"
+
+#include "common/common/assert.h"
+
+namespace Envoy {
+namespace Http {
+namespace Http2 {
+
+MetadataDecoder::MetadataDecoder(uint64_t stream_id) : stream_id_(stream_id) {
+  int rv = nghttp2_hd_inflate_new(&inflater_);
```

Comment:
See comment in enflater about using CSmarePtr for the inflater

Reply:
Great idea! Done.